src/channel_manager.ts

```diff
@@ -0,0 +1,517 @@
+import type { StreamChat } from './client';
+import type {
+  DefaultGenerics,
+  ExtendableGenerics,
+  Event,
+  ChannelOptions,
+  ChannelStateOptions,
+  ChannelFilters,
+  ChannelSort,
+} from './types';
+import { StateStore } from './store';
+import { Channel } from './channel';
+import {
+  findLastPinnedChannelIndex,
+  findPinnedAtSortOrder,
+  getAndWatchChannel,
+  isChannelArchived,
+  isChannelPinned,
+  moveChannelUpwards,
+  shouldConsiderArchivedChannels,
+  shouldConsiderPinnedChannels,
+} from './utils';
+
+export type ChannelManagerPagination<SCG extends ExtendableGenerics = DefaultGenerics> = {
+  filters: ChannelFilters<SCG>;
+  hasNext: boolean;
+  isLoading: boolean;
+  isLoadingNext: boolean;
```

Comment:
Why do we need 2 loading flags?

Reply:
One is for when the entire channel list is loading (i.e `queryChannels`) whereas the other is only when the next page is loading (i.e when `loadNext` is invoked); we need to distinguish between the 2 in my opinion as they technically mean two different things - what do you think ?

Reply:
While it is true that we can (on the SDK's side) distinguish between these 2, if an integrator were to use their own custom `ChannelManager` this information would be useful for them since they wouldn't need to rely on additional logic to figure out the loading state, but rather only on the 2 loading flags. 

`isLoading` basically means a full reload whereas `isLoadingNext` refers to paginating. Also, this opens up the possibility to perhaps depend on one of these specifically in the future if we need to do so, without introducing a breaking change

Reply:
As discussed with Martin on a huddle, we'll keep this behaviour as it can be useful for integrators as well as us